### PR TITLE
Fix single removal actions modals and add some on edit actions

### PR DESCRIPTION
**Pull request by gagnieray**

### CODE REVIEW

File: galette/templates/default/pages/configuration_dynamic_fields.html.twig [hunk 1]
```diff
@@ -77,5 +77,37 @@
...
+                event.preventDefault();
+                var _this = $(this);
+                var _url = _this.attr('href');
+                $.ajax({
+                    url: _url,
+                    type: "GET",
+                    {% include "elements/js/loader.js.twig" %},
+                    success: function(res){
+                        $.modal({
+                            title: '{{ _T("Edit field") }}',
```
Comment (trasher) [inline galette/templates/default/pages/configuration_dynamic_fields.html.twig:91]:
```suggestion
                            title: '{{ _T("Edit field")|e("js") }}',
```

File: galette/templates/default/pages/configuration_entitleds.html.twig [hunk 2]
```diff
@@ -12,7 +12,50 @@
...
         $(function() {
-            {% include 'elements/js/removal.js.twig' %}
+            $('.single-edit').click(function(event){
+                event.preventDefault();
+                var _this = $(this);
+                var _url = _this.attr('href');
+                var _type = _url.split('/').slice(1, 2)[0];
+                switch (_type) {
+                    case 'status':
+                        _title = '{{ _T("Edit status") }}';
```
Comment (trasher) [inline galette/templates/default/pages/configuration_entitleds.html.twig:22]:
```suggestion
                        _title = '{{ _T("Edit status")|e("js") }}';
```

File: galette/templates/default/pages/configuration_entitleds.html.twig [hunk 3]
```diff
@@ -12,7 +12,50 @@
...
+                        break;
+                    case 'contributions-types':
+                        _title = '{{ _T("Edit contribution type") }}';
```
Comment (trasher) [inline galette/templates/default/pages/configuration_entitleds.html.twig:25]:
```suggestion
                        _title = '{{ _T("Edit contribution type")|e("js") }}';
```

File: galette/templates/default/pages/configuration_payment_types.html.twig [hunk 4]
```diff
@@ -84,7 +82,41 @@
...
+                    event.preventDefault();
+                    var _this = $(this);
+                    var _url = _this.attr('href');
+                    $.ajax({
+                        url: _url,
+                        type: "GET",
+                        {% include "elements/js/loader.js.twig" %},
+                        success: function(res){
+                            $.modal({
+                                title: '{{ _T("Edit payment type") }}',
```
Comment (trasher) [inline galette/templates/default/pages/configuration_payment_types.html.twig:95]:
```suggestion
                                title: '{{ _T("Edit payment type")|e("js") }}',
```

File: galette/templates/default/pages/configuration_titles.html.twig [hunk 5]
```diff
@@ -81,7 +81,41 @@
...
+                event.preventDefault();
+                var _this = $(this);
+                var _url = _this.attr('href');
+                $.ajax({
+                    url: _url,
+                    type: "GET",
+                    {% include "elements/js/loader.js.twig" %},
+                    success: function(res){
+                        $.modal({
+                            title: '{{ _T("Edit title") }}',
```
Comment (trasher) [inline galette/templates/default/pages/configuration_titles.html.twig:94]:
```suggestion
                            title: '{{ _T("Edit title")|e("js") }}',
```

**Review by trasher — COMMENTED** [review]
Just a Quick & Dirty review for now

**Comment by gagnieray** [comment]
The missing twig filter should now be properly used when needed. Let me know if you notice any other issue :wink:

**Comment by gagnieray** [comment]
Sorry I didn't notice this PR wasn't in draft. Now, I've finished pushing commits on this feature :smile:

File: galette/lib/Galette/Controllers/DynamicTranslationsController.php [hunk 6]
```diff
@@ -171,8 +173,10 @@ public function doDynamicTranslations(Request $request, Response $response): Res
     {
         $post = $request->getParsedBody();
         $post['text_orig'] = htmlspecialchars($post['text_orig'], ENT_QUOTES);
+        $redirect_url = $post['redirect_uri'];
```
Comment (trasher) [inline galette/lib/Galette/Controllers/DynamicTranslationsController.php:176]: What if the `$post['redirect_uri']` does not exists?

```suggestion
        if(isset($post['redirect_uri'])) {
            $redirect_url = $post['redirect_uri'];
            unnset($post['redurrect_uri']);
        } else {
            $redirect_url = $this->router->pathFor(
                'dynamicTranslations',
                ['text_orig' => $post['text_orig']]
            )
        }
```

And then later use `$redirect_uri` in response.

File: galette/templates/default/pages/configuration_dynamic_translations.html.twig [hunk 7]
```diff
@@ -46,12 +46,17 @@
...
+    {% if mode != 'ajax' %}
             <div class="ui basic center aligned segment">
                 <button type="submit" name="trans" class="ui labeled icon primary button action">
                     <i class="save icon"></i> {{ _T("Save") }}
                 </button>
-                {% include "components/forms/csrf.html.twig" %}
             </div>
+    {% else %}
+            <input type="hidden" name="trans" id="trans" value="true"/>
+            <input type="hidden" name="redirect_uri" id="redirect_uri" value="PATH"/>
```
Comment (trasher) [inline galette/templates/default/pages/configuration_dynamic_translations.html.twig:57]: `PATH` seems wrong here

File: galette/templates/default/pages/configuration_entitleds.html.twig [hunk 8]
```diff
@@ -12,7 +12,21 @@
 {% block javascripts %}
     <script type="text/javascript">
         $(function() {
-            {% include 'elements/js/removal.js.twig' %}
+            {% set type = path_for("doAddEntitled", {"class": url_class})|split("/")|slice(1, 2)[0] %}
```
Comment (trasher) [inline galette/templates/default/pages/configuration_entitleds.html.twig:15]: Hum... I wonder if there is no more clean solution; this seems hard to understand.

File: galette/templates/default/pages/configuration_dynamic_translations.html.twig (hunk 7 above)
Comment (gagnieray) [reply galette/templates/default/pages/configuration_dynamic_translations.html.twig:57]: This condition is for ajax requests. The value should effectively be empty as it it set in js before submit.

File: galette/templates/default/pages/configuration_entitleds.html.twig (hunk 8 above)
Comment (gagnieray) [reply galette/templates/default/pages/configuration_entitleds.html.twig:15]: yes : `{% set type = url_class %}` :rofl:

File: galette/lib/Galette/Controllers/DynamicTranslationsController.php [hunk 9]
```diff
@@ -171,6 +173,15 @@ public function doDynamicTranslations(Request $request, Response $response): Res
     {
         $post = $request->getParsedBody();
         $post['text_orig'] = htmlspecialchars($post['text_orig'], ENT_QUOTES);
+        if(isset($post['redirect_uri'])) {
```
Comment (trasher) [inline galette/lib/Galette/Controllers/DynamicTranslationsController.php:176]:
```suggestion
        if (isset($post['redirect_uri'])) {
```

**Comment by trasher** [comment]
I have just merged the slim4 feature into develop branch. There are many changes, but on Twig templates, the most notable one (and the one that may cause conflicts) is the rename of `path_for` to `url_for`.
There are a few other changes in path resolution that affects JS and CSS icnludes, but I do not think your current PR are impacted.

**Comment by gagnieray** [comment]
> I have just merged the slim4 feature into develop branch. There are many changes, but on Twig templates, the most notable one (and the one that may cause conflicts) is the rename of `path_for` to `url_for`. There are a few other changes in path resolution that affects JS and CSS icnludes, but I do not think your current PR are impacted.

Thank you ! I'll have a look and resolve the conflicts as soon as possible.

**Pull request merged**